src/components/NameFilter.tsx

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable prefer-const */
+import { debounce } from 'lodash';
+import React, { useState, useCallback, useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+export const NameFilter = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const initialQuery = searchParams.get('query') || '';
+  const [query, setQuery] = useState(initialQuery);
+
+  const handleQueryChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setQuery(event.target.value);
+    }, [],
+  );
+
+  const applyQuery = debounce(() => {
```

Comment:
I use the useRef, and call apply Query manually. I think you version is better for any form that doesn't have a submit button. I'll start considering